sos/plugins/watchdog.py

```diff
@@ -0,0 +1,84 @@
+# Copyright (C) 2018 Red Hat, Inc., Reid Wahl <nwahl@redhat.com>
+
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
+
+from sos.plugins import Plugin, RedHatPlugin
+
+import re
+
+
+class Watchdog(Plugin, RedHatPlugin):
+    '''Watchdog information.'''
+    plugin_name = 'watchdog'
+    profiles = ('system',)
+    packages = ('watchdog',)
+    option_list = [
+        ('conf_file', 'watchdog config file', 'fast', '/etc/watchdog.conf'),
+    ]
+
+    def get_log_dir(self):
+        '''Get watchdog log directory.
+
+        Get watchdog log directory path configured in ``conf_file``.
+
+        :returns: The watchdog log directory path.
+        :returntype: str.
+        :raises: IOError if ``conf_file`` is not readable.
+        '''
+        log_dir = None
+        log_dir_pattern = re.compile(r'log-dir\s*=\s*(.*)')
+
+        with open(self.get_option('conf_file'), 'r') as conf_file:
```

Comment:
Why get this twice? If you're implementing a method, then why not simply pass the `watchdog_conf` down from the parent method? (it would also make the reference to ``conf_file`` in the docstring more clear: it looks odd, as it's neither a class member, nor a function argument, but an sos plugin option name).

As a general style thing, it's also good to always use the same name for a variable representing the same thing, unless there's a good reason not to: i.e. I'd suggest either `conf_file` in both places, or `watchdog_conf` (not much to chose between them aside form the shorter length of `conf_file`).

Reply:
Honestly, that's how I had implemented it at first in the revision. And then I figured you would say it was unnecessary/clunky to pass the conf file as an argument to `get_log_dir`, so I switched to this. The coin landed on the wrong side.

Duly noted.

Reply:
Sorry you had to re-do it again - generally, if I don't say anything then something is OK ;-)

Reply:
I would rather you deeply critique everything that comes in, than ack a bunch of stuff that works but looks sloppy or doesn't fit with the project's style.